src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -253,6 +237,32 @@ public override Result VisitResult(Result node)
                 }
             }
 
+            if (_dataToInsert.HasFlag(OptionallyEmittedData.PartialFingerprints))
+            {
+                Location primaryLocation = node.Locations?.FirstOrDefault();
+                if (primaryLocation != null)
+                {
+                    PhysicalLocation physicalLocation = primaryLocation.PhysicalLocation;
+                    Uri resolvedUri = GetResolvedArtifactLocationUri(physicalLocation.ArtifactLocation);
+
+                    Region expandedRegion = _fileRegionsCache.PopulateTextRegionProperties(physicalLocation.Region, resolvedUri, false);
```

Comment:
>Region expandedRegion [](http://example.com/codeflow?start=20&length=21)

Since this code runs after `base.VisitResult(node);`, the region/contextRegion snippets may have already been created.

You can check whether `physicalLocation.Region.Snippet` already exists to avoid computing the value again.

Then this comes with a design question: if `OptionallyEmittedData.PartialFingerprints` is flagged, does it implies `OptionallyEmittedData.RegionSnippets` and `OptionallyEmittedData.ContextRegionSnippets` also are flagged? since to populate the partial fingerprint it has to read the region and context region.

@michaelcfanning what do you think?

Reply:
> design question
I think we should not also emit `ContextRegionSnippets` and `RegionSnippets` because the sarif file will be unnecessarily bloated. The user may want to use the fingerprint only for result matching and do not want to send "Code" to a result management system.

Reply:
@mpoqq had a discussion with the team, this is a valid scenario. Thanks for bringing it up.